ls1intum/Artemis
# Fix Score Calculation for Zero Point Programming Exercises

### Checklist

### Motivation and Context
Calculate the score of zero point programming exercises properly as well (this is common for practice-only exercises that are not graded at all). This worked before we updated the score calculation and is required for PGdP.

**Manual assessment of zero point exercises (programming and in general) are out of scope for this PR. See comment below as well.**

### Description
- Use placeholder point value when calculating the score of a zero point programming exercise. Since it is not used for the students course score (there the exercise score is still 0), the result is only displayed better.
- Adjust some charts in the client to display the values correctly.

### Steps for Testing
0. Review the code.
1. Create a programming exercise with and without SCA but with points.
2. Check that the score calculation still works if you solve test cases (best would be to check weights and bonus points as well)
3. Edit the exercise to give 0 points (and no bonus points as well)
4. Check that the score calculation works just as before (but without you getting any points in the course for that)

### Test Coverage
_Unchanged._
